inc/MediaResponse.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Media Response object.
+ *
+ * @package asset-manager-framework
+ */
+
+declare( strict_types=1 );
+
+namespace AssetManagerFramework;
+
+class MediaResponse {
+
+	/**
+	 * The reponse media list.
+	 *
+	 * @var MediaList
+	 */
+	private $media_list;
+
+	public function __construct( MediaList $media_list, int $total, int $per_page ) {
+		$this->media_list = $media_list;
+
+		// Set pagination headers.
+		if ( ! headers_sent() ) {
+			header( sprintf( 'X-WP-Total: %d', $total ) );
+			header( sprintf( 'X-WP-TotalPages: %d', ceil( $total / $per_page ) ) );
+		}
```

Comment:
Can we please move this outside the constructor? This is a side-effect, and even more one that doesn't have anything to do with the internal data of this class, really.

I suggest to store both `$total` and `$per_page` as fields on this class, and introduce a new method `send_pagination_headers`, which should get called in the `request_items` method of the `Provider` class, right afterthe `$response` has been received (from `$this->request()`).

Reply:
Sure, I did start that way but thought they're only needed here so tried this to be as terse as possible but you're right.